### ISSUE TITLE
Suggestion - update problemmatcher for sjasmplus to filelocation autodetect

### ISSUE THREAD

**Issue by darenm** [mborik/z80-macroasm-vscode#9]
Hi there - thanks for this plugin!
I am setting up my environment and I like to locate my .asm files in a subfolder of the main project solution and although the **$errmatcher-sjasmplus** problem matcher is correctly detecting the error in the assembly output, **sjasmplus** does not include the relative path when displaying the error:

```log
> Executing task: D:\next\DMTestBong\project\bin\sjasmplus d:\next\DMTestBong\project\src\Project.asm --zxnext=cspect --msg=all <


SjASMPlus Z80 Cross-Assembler v1.14.5 (https://github.com/z00m128/sjasmplus)
Pass 1 complete (0 errors)
Pass 2 complete (0 errors)
Project.asm(17): error: Unrecognized instruction: fred a, bill
Pass 3 complete
Errors: 1, warnings: 0, compiled: 57 lines, work time: 0.031 seconds
The terminal process terminated with exit code: 1
```

With the **$errmatcher-sjasmplus** defined as:

```json
{
	"name": "errmatcher-sjasmplus",
	"owner": "z80-macroasm",
	"fileLocation": ["relative", "${workspaceFolder}"],
	"pattern": [
		{
			"regexp": "^(.*)\\(([0-9]+)\\): (error|warning):\\s+(.*)$",
			"file": 1,
			"line": 2,
			"severity": 3,
			"message": 4
		}
	]
},
```
clicking the error in the **Problems** list results in an error as the file is assumed to be relative to the workspace folder.

However, if I amend my build task to include the `--fullpath` switch:

```log
> Executing task: D:\next\DMTestBong\project\bin\sjasmplus d:\next\DMTestBong\project\src\Project.asm --zxnext=cspect --msg=all --fullpath <


SjASMPlus Z80 Cross-Assembler v1.14.5 (https://github.com/z00m128/sjasmplus)
Pass 1 complete (0 errors)
Pass 2 complete (0 errors)
d:\next\DMTestBong\project\src\Project.asm(17): error: Unrecognized instruction: fred a, bill
Pass 3 complete
Errors: 1, warnings: 0, compiled: 57 lines, work time: 0.015 seconds
The terminal process terminated with exit code: 1

Press any key to close the terminal.
```

This time, clicking the error in the **Problems** list results in an error as the file is still assumed to be relative to the workspace folder.

However, if I edit the extension **package.json** to use "autodetect":

```json
{
	"name": "errmatcher-sjasmplus",
	"owner": "z80-macroasm",
	"fileLocation": ["autodetect", "${workspaceFolder}"],
	"pattern": [
		{
			"regexp": "^(.*)\\(([0-9]+)\\): (error|warning):\\s+(.*)$",
			"file": 1,
			"line": 2,
			"severity": 3,
			"message": 4
		}
	]
},
```

The problem path works whether the .asm file is in the workspace folder or nested lower.

**Comment by mborik** [comment]
Hi, thanks for suggestion. Autodetect is probably new feature in problem matchers, it’s grear idea, will be changed. 😉

**Comment by mborik** [comment]
fixed in 9a826e9
